### PR TITLE
Add support for darwin/amd64 and windows/amd64

### PR DESCRIPTION
**Pull request by zhuyie**

### CODE REVIEW

**Comment by zhuyie** [comment]
Looks like Travis CI not support MS Windows right now...
https://docs.travis-ci.com/user/reference/overview/#Virtualisation-Environment-vs-Operating-System

File: linkzstd_darwin.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package gozstd
```
Comment (valyala) [inline linkzstd_darwin.go]: could you rename files starting from `link*` to `lib*`, since they are related to `libzstd` library?

Comment (zhuyie) [reply linkzstd_darwin.go]: Done.

Comment (zhuyie) [reply linkzstd_darwin.go]: Since you need gcc to use cgo on Windows, then you can install MSYS2 to run the Makefile.

**Comment by valyala** [comment]
@zhuyie , thanks for the contribution . I didn't test MacOS and Windows builds, but assume it works

**Comment by valyala** [comment]
@zhuyie , could you re-run `make clean libzstd.a` under Windows and send pull request with the updated `*a` file?

**Pull request merged**